src/main/kotlin/app/revanced/patches/youtube/misc/quality/patch/DefaultVideoQualityPatch.kt

```diff
@@ -0,0 +1,79 @@
+package app.revanced.patches.youtube.misc.quality.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstruction
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.annotations.Dependencies
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.fingerprint.method.utils.MethodFingerprintUtils.resolve
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.quality.annotations.DefaultVideoQualityCompatibility
+import app.revanced.patches.youtube.misc.quality.fingerprints.VideoUserQualityChangeFingerprint
+import app.revanced.patches.youtube.misc.quality.fingerprints.VideoQualityReferenceFingerprint
+import app.revanced.patches.youtube.misc.quality.fingerprints.VideoQualitySetterFingerprint
+import app.revanced.patches.youtube.misc.videoid.fingerprint.VideoIdFingerprint
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.reference.FieldReference
+
+@Patch(false)
+@Dependencies(
+    dependencies = [IntegrationsPatch::class]
+)
+@Name("default-video-quality")
+@Description("Adds the ability to select preferred video quality.")
+@DefaultVideoQualityCompatibility
+@Version("0.0.1")
+class DefaultVideoQualityPatch : BytecodePatch(
+    listOf(
+        VideoQualitySetterFingerprint,
+        VideoIdFingerprint
+    )
+

```

Comment:
Do not forget to apply auto formatting.

Reply:
Well i did a thing.
Don't know if it's what you wanted.